### PR TITLE
List Leaves Command

### PR DESCRIPTION
**Pull request by 4ndrelim**
**IMPORTANT**

While implementing the new feature, i realised equals::ModelManager was not updated prviously, leading to some test cases to pass erroneously. After editing, further test files showed failure:
1. `ListEmployeesInDepartmentCommandTest`
2. `EditDepartmentCommandTest`
3. `ListEmployeeDepartmentCommandTest`

I haven't looked too much into it but the fails likely occur because the edit commands fail to account for the filtered list view. I suspect there might not be anything wrong with implementation, just incorrect initialization of testing variables but pls do take a look since y'all will know better.

Hate to throw back to y'all but spent quite abit of time code tracing to figure out the previous error and ps need do other stuff. I'll follow-up on this agn tmr night

### CODE REVIEW

File: src/test/java/seedu/sudohr/logic/commands/department/ListEmployeesInDepartmentCommandTest.java [hunk 1]
```diff
@@ -42,6 +42,7 @@ public void execute_unfilteredList_success() {
 
     }
 
+    // BUGGED
     @Test
     public void execute_filteredList_success() {
```
Comment (4ndrelim) [inline src/test/java/seedu/sudohr/logic/commands/department/ListEmployeesInDepartmentCommandTest.java:47]: This method is bugged

File: src/test/java/seedu/sudohr/logic/commands/department/ListEmployeeDepartmentCommandTest.java [hunk 2]
```diff
@@ -26,10 +24,13 @@ public void setUp() {
         expectedModel = new ModelManager(model.getSudoHr(), new UserPrefs());
     }
 
+    // BUGGED
     @Test
     public void execute_listIsFiltered_showsSameList() {
```
Comment (4ndrelim) [inline src/test/java/seedu/sudohr/logic/commands/department/ListEmployeeDepartmentCommandTest.java:29]: bugged

File: src/test/java/seedu/sudohr/logic/commands/department/EditDepartmentCommandTest.java [hunk 3]
```diff
@@ -33,6 +33,7 @@ public class EditDepartmentCommandTest {
 
     private Model model = new ModelManager(getTypicalSudoHr(), new UserPrefs());
 
+    // BUGGED
     @Test
     public void execute_allFieldsSpecifiedUnfilteredList_success() {
```
Comment (4ndrelim) [inline src/test/java/seedu/sudohr/logic/commands/department/EditDepartmentCommandTest.java:38]: bugged

File: src/main/java/seedu/sudohr/model/ModelManager.java [hunk 4]
```diff
@@ -352,7 +352,8 @@ public boolean equals(Object obj) {
         ModelManager other = (ModelManager) obj;
         return sudoHr.equals(other.sudoHr)
                 && userPrefs.equals(other.userPrefs)
-                && filteredEmployees.equals(other.filteredEmployees);
+                && filteredEmployees.equals(other.filteredEmployees)
+                && filteredDepartments.equals(other.filteredDepartments)
```
Comment (4ndrelim) [inline src/main/java/seedu/sudohr/model/ModelManager.java:356]: Here's prob the changes that caused the bug

**Review by kennethk-1201 — APPROVED** [review]
LGTM

**Pull request merged**